### ISSUE TITLE
nfl.com

### ISSUE THREAD

**Issue by luixxiul** [brave/browser-laptop#3137]
From support:

> I have an arrow pointing to a row that features scores of many NFL games. The brave browser does not show this, unless I allow ads. I would prefer to block ads though. Maybe this is how the http://www.nfl.com/ website is constructed
> 
> ![clipboard01](https://cloud.githubusercontent.com/assets/3362943/17611034/7cd890f8-607e-11e6-9052-0917c2f4fc92.png)

**Comment by lukemulks** [comment]
Hi @luixxiul - I can speak to this one w/some experience (I used to work directly with the NFL). 

There are two potential blocking items that prevent the gamecenter icons, countdown clock and stats counter from displaying in the field that you've highlighted above. I'm going to cover each of them below - and I suspect that either a siteHack or some other custom work would need to take place to fully resolve. 

**Temporarily - as you noted - having shields down will allow for the content to fully load.** 

OK - so the NFL has the primary CDN that their site is hosted from for nfl.com, but also has content served and hosted from some additional CDNs. 

NFL gameday stats, countdown clock, and other modular content that fits in the area you highlighted is served via: 

- `i.nflcdn.com` (for iframe and other interactive content) 
- `s.nflcdn.com` (for scripts) 

NFL also uses Adobe (domains: `auditude`, `omniture`, `demdex`) for video on demand (VOD) and live streaming video content for their properties. (I've filed a ticket for enabling Adobe Primetime DRM, which may help with some of their authenticated content if they're using Primetime DRM as a feature - which they may be doing now). 

We block this script that NFL uses for their doubleclick ads: `http://s.nflcdn.com/static/site/7.4/scripts/ad.js`

We should continue blocking that script. 

I suspect that the content in question does not load from the `http://i.nflcdn.com` domain because those requests use `document.write` for inserting the content (which our Shields block). IIRC, `document.write`  is allowed with Shields down (but I could be wrong with that assumption).

I'm not sure if a workaround would potentially be a `siteHack` to replace `document.write`  for `innerHTML` (or something along those lines). Not sure how easy/hard that would be to implement, but the notes above are likely why this issue is occurring for NFL.com.
